bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/tools/ToolsInstallationJob.java

```diff
@@ -1,161 +0,0 @@
-import org.eclipse.core.runtime.IProgressMonitor;
-import org.eclipse.core.runtime.IStatus;
-import org.eclipse.core.runtime.Status;
-import org.eclipse.ui.console.MessageConsoleStream;
-
-import com.espressif.idf.core.IDFConstants;
-import com.espressif.idf.core.IDFCorePlugin;
-import com.espressif.idf.core.IDFEnvironmentVariables;
-import com.espressif.idf.core.logging.Logger;
-import com.espressif.idf.core.toolchain.ESPToolChainManager;
-import com.espressif.idf.core.toolchain.ESPToolchain;
-import com.espressif.idf.core.tools.vo.IDFToolSet;
-import com.espressif.idf.ui.update.ExportIDFTools;
-import com.espressif.idf.ui.update.Messages;
-
-/**
- * Class to install the dependencies and any 
- * given tools for an idf with given {@link IDFToolSet}
- * @author Ali Azam Rana
- *
- */
-public class ToolsInstallationJob extends ToolsJob
-{
-	public ToolsInstallationJob(String pythonExecutablePath, String gitExecutablePath, String idfPath)
-	{
-		super(Messages.InstallToolsHandler_InstallingToolsMsg, pythonExecutablePath, gitExecutablePath);
-		this.idfPath = idfPath;
-		this.idfToolSet.setIdfLocation(idfPath);
-		this.idfToolSet.setSystemGitExecutablePath(gitExecutablePath);
-		this.idfToolSet.setSystemPythonExecutablePath(pythonExecutablePath);
-	}
-
-	@Override
-	protected IStatus run(IProgressMonitor monitor)
-	{
-		monitor.beginTask(Messages.InstallToolsHandler_ItWilltakeTimeMsg, 5);
-		monitor.worked(1);
-		Logger.log(""
-				);
-		IStatus status = handleToolsInstall();
-		if (status.getSeverity() == IStatus.ERROR)
-		{
-			return status;
-		}
-
-		monitor.worked(1);
-		monitor.setTaskName(Messages.InstallToolsHandler_InstallingPythonMsg);
-		status = handleToolsInstallPython(console);
-		if (status.getSeverity() == IStatus.ERROR)
-		{
-			return status;
-		}
-
-		monitor.worked(1);
-
-		monitor.setTaskName(Messages.InstallToolsHandler_ExportingPathsMsg);
-		ExportIDFTools exportIDFTools = new ExportIDFTools();
-		status = exportIDFTools.getToolsExportOutputFromGivenIdfPath(pythonExecutablePath, gitExecutablePath, console,
-				errorConsoleStream, idfPath);
-		if (status.getSeverity() == IStatus.ERROR)
-		{
-			return status;
-		}
-		processExportCmdOutput(status.getMessage());
-
-		monitor.worked(1);
-
-		monitor.setTaskName(Messages.InstallToolsHandler_AutoConfigureToolchain);
-		ESPToolChainManager espToolChainManager = new ESPToolChainManager();
-		String pathToLookForToolChains = idfToolSet.getEnvVars().get(IDFEnvironmentVariables.PATH);
-		String idfPath = idfToolSet.getEnvVars().get(IDFEnvironmentVariables.IDF_PATH);
-		try
-		{
-			List<ESPToolchain> espToolChains = espToolChainManager
-					.getStdToolChains(Arrays.asList(pathToLookForToolChains.split(File.pathSeparator)), idfPath);
-			idfToolSet.setEspStdToolChains(espToolChains);
-			List<ICMakeToolChainFile> cMakeToolChainFiles = espToolChainManager.getCmakeToolChains(idfPath);
-			idfToolSet.setEspCmakeToolChainFiles(cMakeToolChainFiles);
```

Comment:
As we need tools configured after activation, are we handling this somewhere?

Reply:
@kolipakakondal yes create a separate class to avoid merge conflicts as this work can take some time. All this is handled properly